<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Когда-используется-критерий-Стьюдента,-а-когда-Z-–критерий?-Задачи-2,3-решать-вручную" data-toc-modified-id="Когда-используется-критерий-Стьюдента,-а-когда-Z-–критерий?-Задачи-2,3-решать-вручную-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Когда используется критерий Стьюдента, а когда Z –критерий? Задачи 2,3 решать вручную</a></span></li><li><span><a href="#Проведите-тест-гипотезы.-Утверждается,-что-шарики-для-подшипников,-изготовленные-автоматическим-станком,-имеют-средний-диаметр-17-мм.-Используя-односторонний-критерий-с-α=0,05,-проверить-эту-гипотезу,-если-в-выборке-из-n=100-шариков-средний-диаметр-оказался-равным-17.5-мм,-а-дисперсия-известна-и-равна-4-кв.-мм." data-toc-modified-id="Проведите-тест-гипотезы.-Утверждается,-что-шарики-для-подшипников,-изготовленные-автоматическим-станком,-имеют-средний-диаметр-17-мм.-Используя-односторонний-критерий-с-α=0,05,-проверить-эту-гипотезу,-если-в-выборке-из-n=100-шариков-средний-диаметр-оказался-равным-17.5-мм,-а-дисперсия-известна-и-равна-4-кв.-мм.-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.</a></span></li><li><span><a href="#Проведите-тест-гипотезы.-Продавец-утверждает,-что-средний-вес-пачки-печенья-составляет-200-г.-Из-партии-извлечена-выборка-из-10-пачек.-Вес-каждой-пачки-составляет:-202,-203,-199,-197,-195,-201,-200,-204,-194,-190.-Известно,-что-их-веса-распределены-нормально.-Верно-ли-утверждение-продавца,-если-учитывать,-что-доверительная-вероятность-равна-99%?-(Провести-двусторонний-тест.)" data-toc-modified-id="Проведите-тест-гипотезы.-Продавец-утверждает,-что-средний-вес-пачки-печенья-составляет-200-г.-Из-партии-извлечена-выборка-из-10-пачек.-Вес-каждой-пачки-составляет:-202,-203,-199,-197,-195,-201,-200,-204,-194,-190.-Известно,-что-их-веса-распределены-нормально.-Верно-ли-утверждение-продавца,-если-учитывать,-что-доверительная-вероятность-равна-99%?-(Провести-двусторонний-тест.)-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)</a></span></li><li><span><a href="#Есть-ли-статистически-значимые-различия-в-росте-дочерей?-Рост-матерей-172,-177,-158,-170,-178,175,-164,-160,-169,-165-Рост-взрослых-дочерей:-173,-175,-162,-174,-175,-168,-155,-170,-160" data-toc-modified-id="Есть-ли-статистически-значимые-различия-в-росте-дочерей?-Рост-матерей-172,-177,-158,-170,-178,175,-164,-160,-169,-165-Рост-взрослых-дочерей:-173,-175,-162,-174,-175,-168,-155,-170,-160-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Есть ли статистически значимые различия в росте дочерей? Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165 Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160</a></span></li></ul></div>

## Когда используется критерий Стьюдента, а когда Z –критерий? Задачи 2,3 решать вручную

Z критерий применяем когда известна дисперсия генеральной совокупности.

Т критерий Стьюдента применяется когда дисперсия неизвестна.

## Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [1]:
D = 4
std = D ** 0.2
M = 17
X = 17.5
alpha = 0.05
n = 100

Дисперсия известна, используем Z критерий

In [2]:
Z = (X - M) / (std * (n ** 0.5))
Z

0.03789291416275995

In [3]:
Z_alpha = 1.645 # По таблице

Z < Z_alpha , значит p-value будет больше альфа. Принимаем Н0, средний диаметр может быть 17мм.

p-value будет 0.51 по таблице.

## Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.) 

In [4]:
x = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
n = 10
M = 200
alpha = 0.01

Дисперсия неизвестна, проводит Т-тест, причем альфа нужно брать 0.5% с каждой стороны.

Найдем среднее по выборке

In [5]:
x_mean = sum(x) / len(x)
x_mean

198.5

Найдем дисперсию и среднее квадратичное отклонение для выборки(несмещенные)

In [6]:
D = 0
for i in range(len(x)):
    D += (x[i] - x_mean) ** 2
D = D / (len(x) - 1)
D

19.833333333333332

In [7]:
std = D ** 0.5
std

4.453463071962462

Найдем Т критерий

In [8]:
t = (x_mean - M) / (std * (n ** 0.5))
t

-0.10651074037450896

t_alpha критерий для уровня значимости 1% равен 2,6258905

Мы получили Т критерий меньше, значит оставляем Н0, средний вес равен 200гр.

## Есть ли статистически значимые различия в росте дочерей? Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165 Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [9]:
x_1 = [172, 177, 158, 170, 178, 175, 164, 160, 169, 165]
x_2 = [173, 175, 162, 174, 175, 168, 155, 170, 160]

Я понимаю, что у матерей бывает по 1, 2 и более дочек. Это нормально

Но ситуация когда у дочки может быть не 1, а 2 или более матерей выглядит как минимум странно.

Несмотря на это проведем тест Шаприро, для понимания распределений. И далее, если распределения нормальные, мы должны провести партый Т-тест для зависимых выборок, находить дисперсии распределений в этом случае даже не требуется.

Однако, выборки имеют разное число наблюдений, поэтому парный Т-тест не применим. Будем делать двухвыборочный тест для независимых выборок. И для этого нам потребуется сравнить дисперсии.

In [10]:
from scipy import stats

In [11]:
stats.shapiro(x_1)

ShapiroResult(statistic=0.9526640176773071, pvalue=0.7000676393508911)

In [12]:
stats.shapiro(x_2)

ShapiroResult(statistic=0.879397988319397, pvalue=0.15474608540534973)

Примем уровень значимости за 0.05, и при нём обе выборки имеют нормальное распределение.

Проверим равенство дисперсий тестом Левена

In [13]:
stats.levene(x_1, x_2)

LeveneResult(statistic=0.008702276949164091, pvalue=0.926766959806049)

Принимаем Н0 о равенстве дисперсий. Проведем т-тест.

In [14]:
stats.ttest_ind(x_1, x_2)

Ttest_indResult(statistic=0.24400587215969016, pvalue=0.8101467126543231)

pvalue составляет 81%, что значимо больше 5% принятого уровня значимости.

Принимаем гипотезу Н0, рост матерей и дочерей не имеет статистически значимой разницы.